packages/web-components/@sfx/sayt/src/sayt.ts

```diff
@@ -182,9 +203,31 @@ export default class Sayt extends LitElement {
       this.hideSayt();
       return;
     }
+    this.debouncedRequestSaytAutocompleteTerms(query, searchbox);
+    this.debouncedRequestSaytProducts(query, searchbox);
+  }
 
-    this.requestSaytAutocompleteTerms(query, searchbox);
-    this.requestSaytProducts(query, searchbox);
+  /**
+   * Debounces the function that is passed in the callback paramater.
+   * Customizable delay based on debounceTime attribute on Sayt element.
+   *
+   * @param callback The function to debounce.
+   */
+  getDebounce<requestSaytType extends Function>(callback: requestSaytType): ReturnType<debounce> {
+    return debounce(
+      callback,
+      this.debounce,
+      false
+    );
+  }
+
+  /**
+   * Dispatches getDebounce for each of the methods to be debounced.
+   * Is triggered once in constructor and when the debounce attribute is changed.
+   */
+  setDebounce() {
+    this.debouncedRequestSaytAutocompleteTerms  = this.getDebounce(this.requestSaytAutocompleteTerms);
+    this.debouncedRequestSaytProducts  = this.getDebounce(this.requestSaytProducts);
```

Comment:
Since `getDebounce` is only being used in this function, I don't think we need to make it a separate method. We can use `debounce` directly:

```js
    this.debouncedRequestSaytAutocompleteTerms = debounce(this.requestSaytAutocompleteTerms, this.debounce, false);
    this.debouncedRequestSaytProducts = debounce(this.requestSaytProducts, this.debounce, false);
```